packages/sdk/src/subscription_builder_impl.ts

```diff
@@ -65,26 +66,192 @@ export class SubscriptionBuilderImpl<
    */
   onError(
     cb: (ctx: ErrorContextInterface<DBView, Reducers, SetReducerFlags>) => void
-  ): SubscriptionBuilderImpl {
+  ): SubscriptionBuilderImpl<DBView, Reducers, SetReducerFlags> {
     this.#onError = cb;
     return this;
   }
 
   /**
-   * Issues a new `Subscribe` message,
-   * adding `query` to the client's set of subscribed queries.
+   * Subscribe to a single query. The results of the query will be merged into the client
+   * cache and deduplicated on the client.
    *
-   * `query` should be a single SQL `SELECT` statement.
+   * @param query_sql A `SQL` query to subscribe to.
    *
-   * Installs the above callbacks into the new `SubscriptionHandle`,
-   * before issuing the `Subscribe` message, to avoid race conditions.
+   * @example
    *
-   * Consumes the `SubscriptionBuilder`,
-   * because the callbacks are not necessarily `Clone`.
+   * ```ts
+   * const subscription = connection.subscriptionBuilder().onApplied(() => {
+   *   console.log("SDK client cache initialized.");
+   * }).subscribe("SELECT * FROM User");
    *
-   * @param query_sql - The SQL query to subscribe to.
+   * subscription.unsubscribe();
+   * ```
    */
-  subscribe(query_sql: string[]): void {
-    this.db['subscribe'](query_sql, this.#onApplied, this.#onError);
+  subscribe(
+    query_sql: string
+  ): SubscriptionHandleImpl<DBView, Reducers, SetReducerFlags> {
+    return new SubscriptionHandleImpl(
+      this.db,
+      query_sql,
+      this.#onApplied,
+      this.#onError
+    );
+  }
+
+  /**
+   * Subscribes to all rows from all tables.
+   *
+   * This method is intended as a convenience
+   * for applications where client-side memory use and network bandwidth are not concerns.
+   * Applications where these resources are a constraint
+   * should register more precise queries via `subscribe`
+   * in order to replicate only the subset of data which the client needs to function.
+   *
+   * This method should not be combined with `subscribe` on the same `DbConnection`.
+   * A connection may either `subscribe` to particular queries,
+   * or `subscribeToAllTables`, but not both.
+   * Attempting to call `subscribe`
+   * on a `DbConnection` that has previously used `subscribeToAllTables`,
+   * or vice versa, may misbehave in any number of ways,
+   * including dropping subscriptions, corrupting the client cache, or throwing errors.
+   */
+  subscribeToAllTables(): void {
+    this.subscribe('SELECT * FROM *');
```

Comment:
Does this work? I would expect it not to, since `this.subscribe` likely sends a "new-style" `SubscribeSingle`, where `select * from *` only works with the "old-style" `Subscribe`.

Reply:
That is a good catch. I don't think I realized that, but it makes sense.

Reply:
I think we'll need a follow up PR. Unfortunately I have to switch on to other stuff, so we'll have to assign it to someone else. I filed https://github.com/clockworklabs/SpacetimeDB/issues/2245 to track it.